tests/infra/node.py

```diff
@@ -254,6 +255,8 @@ def _read_ports(self):
 
     def stop(self):
         if self.remote and self.network_state is not NodeNetworkState.stopped:
+            if self.suspended:
```

Comment:
Why?

Reply:
We terminate with `SIGTERM`, which is ignored if the process is currently suspended, so if the infra tries to teardown the network (because something failed) while a node is suspended, it will hang waiting for a node termination that never happens. There may be a watchdog timeout we can set elsewhere, but this seems sufficient.